utils/stringcomparator/stringcomparator.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package stringcomparator
+
+type StringComparator struct {
```

Comment:
Public artefacts need a comment.
Sane with New below.

Reply:
Thanks Ian will update this :)